mass/adapters/outbound/dao.py

```diff
@@ -64,3 +67,27 @@ def get_dao(self, *, class_name: str) -> ResourceDao:
             return self._resource_daos[class_name]
         except KeyError as err:
             raise DaoNotFoundError(class_name=class_name) from err
+
+    def collection_init_and_index_creation(self) -> None:
+        # get client
+        client: MongoClient = MongoClient(
+            self._config.db_connection_str.get_secret_value()
+        )
+        db = client[self._config.db_name]
+
+        expected_collections = list(self._config.searchable_classes)
+        existing_collections = db.list_collection_names()
```

Comment:
```suggestion
        expected_collections = self._config.searchable_classes
        existing_collections = set(db.list_collection_names())
```

You can directly iterate over a dict, no conversion to list needed. For the "not in" tests conversion to a set is faster. Though of course in practice this doesn't make a difference.

Reply:
Changed in [3a4aad1](https://github.com/ghga-de/mass/pull/13/commits/3a4aad17154b5abca415b9c13cff8a23282aead6)